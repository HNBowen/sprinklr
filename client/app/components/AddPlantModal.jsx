import React from 'react'

const AddPlantModal = (props) => {
  return (
    <div>
      <form>
        Name: <input type="text" name="name" data-test-id="plantName"/>
        Image: <input type="fileInput" name="image" data-test-id="plantImage"/>
        <button type="submit">Add</button>
      </form>
    </div>
  )
}

export default AddPlantModal