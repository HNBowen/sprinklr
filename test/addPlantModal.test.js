import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import AddPlantModal from '../client/app/components/AddPlantModal.jsx'

describe('AddPlantModal', function() {
  it ('should render correctly', function() {
    const wrapper = shallow(<AddPlantModal/>);
    jestExpect(shallowToJson(wrapper)).toMatchSnapshot();
  })

  //it should render with an input field for the name
  it('should render with an input field for the name', function() {
    const wrapper = shallow(<AddPlantModal/>);
    expect(wrapper.find('[data-test-id="plantName"]')).to.have.length(1);
  })
  //it should render with an image upload field
  it('should render wtih an image upload field', function() {
    const wrapper = shallow(<AddPlantModal/>);
    expect(wrapper.find('[data-test-id="plantImage"]')).to.have.length(1)
  })
  //it should render with a submit button
  it('sould render with a submit button', function() {
    const wrapper = shallow(<AddPlantModal/>);
    expect(wrapper.find('button')).to.have.length(1);
  })
  //it should call addPlant function on submit
  
})