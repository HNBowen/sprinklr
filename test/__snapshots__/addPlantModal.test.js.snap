// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlantModal should render correctly 1`] = `
<div>
  <form>
    Name: 
    <input
      data-test-id="plantName"
      name="name"
      type="text"
    />
    Image: 
    <input
      data-test-id="plantImage"
      name="image"
      type="fileInput"
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
